src/index.js

```diff
@@ -164,7 +165,76 @@ class PubgApi {
     * @returns A Promise with the result or an error
     */
     this.loadMatchById = (matchId, shard = this.defaultShard) =>
-      this.wrapAsync(this.requestAPI(shard, this.routesURI.matches + matchId));
+      return this.wrapAsync(this.requestAPI(shard, this.routesURI.matches + matchId));
+
+
+    /**
+    * Need to wrap Telemetry Functions for RxJS
+    */
+
+    /**
+    * Queries an object of match data(s) for the Telemetry URL - or multiple if provided with multiple matches
+    *
+    * https://developer.playbattlegrounds.com/docs/en/telemetry.html#telemetry-events
+    *
+    * @param {object} parsedData - An object with match data attained through loadMatches, loadMatchByID or otherwise provided
+    *
+    * @returns A Promise with the result or an error
+    */
+    this.findTelemetryURLs = (parsedData) => new Promise((resolve, reject) => {
+      setTimeout(() => {
```

Comment:
So I've actually been toying with ways to handle this. Set Timeout of 0 just puts the event at the end of the message queue, that way it is called asynchronously. However, .then() handlers are wrapped in promises and asynchronous by default. It is possible that users won't follow the guidelines for accessing data though, or have custom use cases. At which point I still wanted this function to be asynch, primarily because JSON parsing can be a lengthier request that a network call and could block other code. https://developer.mozilla.org/en-US/docs/Web/JavaScript/EventLoop

Reply:
Because the API only ever returns a fixed number of matches at a time, this function should only ever have to parse a JSON object of a fixed length and so it will be constant time. However, there may be erroneous use cases where the user has manipulated the data in some way and clogs up the function with a huge JSON object. So I wanted to make sure it doesn't stall the server. I may update this even further using Process.NextTick() after each loop iteration in order to ensure it doesn't block.